crates/zksync/core/src/vm/inspect.rs

```diff
@@ -319,7 +324,11 @@ where
 
     let mut storage: rHashMap<Address, rHashMap<rU256, StorageSlot>> = Default::default();
     let mut codes: rHashMap<Address, (B256, Bytecode)> = Default::default();
-    for (k, v) in &modified_storage {
+    for (k, v) in modified_storage.iter().filter(|(k, _)| {
+        !(k.address() == &NONCE_HOLDER_ADDRESS &&
+            should_ignore_nonce_update &&
+            get_nonce_key(&initiator_address) == **k)
+    }) {
```

Comment:
If the backend was cloned then no changes would be persisted in setups at all...
This is an edge case where the initiator's (`tx.origin`) tx nonce is increased in each of the calls/create that we run with EraVM, but in EVM that does not happen.
If the `msg.sender` is the same as `tx.origin` then the call/create is from `tx.origin` (and not thru a contract) so the nonce change is persisted.

The `is_static` is also a precedent for editing `modified_storage` here, and I would consider it a bug since each call/create is its own tx, while in EVM it isn't